### PR TITLE
Use pragma to avoid -Werror=unused-function on signals

### PR DESCRIPTION
**Pull request by Diego-OT**
Avoiding compilation errors in cases where the -Werror flag is used

### Pull Request Prelude

- [ ] I have followed [proper The Forgotten Server code styling][code].
- [ ] I have read and understood the [contribution guidelines][cont] before making this PR.
- [ ] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

* fix the compilation error for those who use the -Werror flag

" error: ‘void {anonymous}::sigbreakHandler()’ defined but not used [-Werror=unused-function]"

https://prnt.sc/1u3hn16

**Issues addressed:** 

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

File: src/signals.cpp [hunk 1]
```diff
@@ -172,7 +176,7 @@ void dispatchSignalHandler(int signal)
 			// hold the thread until other threads end
 			g_scheduler.join();
 			g_databaseTasks.join();
-			g_dispatcher.join();
+			g_dispatcher.join();
```
Comment (EPuncker) [inline src/signals.cpp:179]:
```suggestion
			g_dispatcher.join();
```

**Comment by ranisalt** [comment]
Huh, why aren't they used?

**Review by DSpeichert — CHANGES_REQUESTED** [review]
See earlier comment.

This also clearly causes warnings.

**Comment by DSpeichert** [comment]
Closing because abandoned.

**Pull request closed without merging**